src/components/AssetSearch/AssetSearch.tsx

```diff
@@ -61,78 +73,129 @@ export const AssetSearch: FC<AssetSearchProps> = ({
     [history],
   )
   const handleClick = onClick ?? defaultClickHandler
-  const [searchTermAssets, setSearchTermAssets] = useState<Asset[]>([])
   const { register, watch } = useForm<{ search: string }>({
     mode: 'onChange',
     defaultValues: {
       search: '',
     },
   })
-  const searchString = watch('search')
+  const searchString = watch('search').trim()
   const searching = useMemo(() => searchString.length > 0, [searchString])
-  useEffect(() => {
-    if (filteredAssets) {
-      setSearchTermAssets(
-        searching ? filterAssetsBySearchTerm(searchString, filteredAssets) : filteredAssets,
-      )
-    }
-  }, [searchString, searching, filteredAssets])
-  const listAssets = searching ? searchTermAssets : filteredAssets
-  /**
-   * display a list of chain icon filters, based on a unique list of chain ids,
-   * derived from the output of the filterBy function, sorted by market cap
-   */
-  const filteredChainIdsByMarketCap: ChainId[] = useMemo(
-    () => intersection(chainIdsByMarketCap, uniq(assets.map(a => a.chainId))),
-    [chainIdsByMarketCap, assets],
-  )
+
+  const chainIds: ChainId[] = useMemo(() => ['All', ...uniq(assets.map(a => a.chainId))], [assets])
   const inputProps: InputProps = useMemo(
     () => ({
       ...register('search'),
       type: 'text',
       placeholder: translate('common.searchAsset'),
       pl: 10,
       variant: 'filled',
+      borderWidth: 0,
       autoComplete: 'off',
       autoFocus: true,
     }),
     [register, translate],
   )
-  const handleChainClick = useCallback(
-    (e: React.MouseEvent) => (chainId: ChainId | 'All') => {
-      e.preventDefault()
-      return setActiveChain(chainId)
-    },
-    [],
-  )
 
   const handleSubmit = useCallback((e: FormEvent<unknown>) => e.preventDefault(), [])
 
+  const popularAssets = useMemo(() => {
+    return popularAssetsByChainId?.[activeChain] ?? []
+  }, [activeChain, popularAssetsByChainId])
+
```

Comment:
Could extract this and the below to react-query selectors possibly to abstract this away

Reply:
could do, but is it necessary?